packages/workspace/src/utils/rules/workspace.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { Tree } from '@angular-devkit/schematics';
+import { UnitTestTree } from '@angular-devkit/schematics/testing';
+import { readJsonInTree, updateJsonInTree } from '../ast-utils';
+import { callRule } from '../testing';
+import { setDefaultCollection } from './workspace';
+import {
+  _test_addWorkspaceFile,
+  WorkspaceFormat
+} from '@angular-devkit/core/src/workspace/core';
+
+describe('Workspace', () => {
+  const defaultCollectionName = '@nrwl/node';
+  const workspaceJsonFileName = 'workspace.json';
+  const nrwlWorkspaceName = '@nrwl/workspace';
+  let tree: UnitTestTree;
+
+  beforeEach(() => {
+    _test_addWorkspaceFile('workspace.json', WorkspaceFormat.JSON);
```

Comment:
Why is this necessary?

Reply:
If the `workspace.json` is not added as workspace file the `readWorkspace` method (used by `updateWorkspace`) throws an error ("Unable to locate a workspace file for workspace path") as (per default) it only searches for `angular.json` or `.angular.json` in the workspace as workspace file.

As the tests uses the `workspace.json` as test file it is required to register it as a workspace file. If this should be avoided the test would have to be based on a `angular.json` file. I somehow mirrored the approach of the `createEmptyWorkspace` method which also used the `workspace.json` instead of the `angular.json` (probably because the latter requires more work to set up).

An alternative could be to extend these methods to take a kind of loader/reader function which then replaces the `readWorkspace`  method from angular-devkit. However this would also require to make changes to the `updateWorkspace` method (I did not want to touch these "core" methods as I don't know if you want them to be stable/not changed).